sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -173,7 +200,33 @@ class GpuSortAggregateMeta(
         case "partial" => if (hashAggMode.contains(Final) || hashAggMode.contains(Complete)) {
           // replacing only Partial hash aggregates, so a Final or Commplete one should not replace
           willNotWorkOnGpu("Replacing Final or Complete hash aggregates disabled")
-         }
+        }
+          // In partial mode, if there are non distinct functions and multiple distinct functions
+          // non distinct functions are computed using First operator. The final result would be
+          // incorrect for non distinct functions for partition size > 1. Reason for this is - if
+          // the first batch computed and sent to CPU doesn't contain all the rows required to
+          // compute non distinct function(s), then Spark would consider that value as final result
+          // (due to First)Falling back to CPU for this special case.
```

Comment:
Nit : non distinct -> non-distinct
Period after (due to First) and some rephrasing.

Reply:
@nartal1 ^

Reply:
Sorry, missed this.